optaplanner-examples/src/main/java/org/optaplanner/examples/common/experimental/impl/ConsecutiveIntervalInfoImpl.java

```diff
@@ -0,0 +1,282 @@
+        removeSpannedBreaksAndUpdateIntersectedBreaks(interval, firstIntersectedIntervalCluster);
+
+        // If the first intersected interval cluster start after the interval,
+        // we need to make the interval start point the key for this interval
+        // cluster in the map
+        if (oldStart.isAfter(firstIntersectedIntervalCluster.getStartSplitPoint())) {
+            clusterStartSplitPointToCluster.remove(oldStart);
+            clusterStartSplitPointToCluster.put(firstIntersectedIntervalCluster.getStartSplitPoint(),
+                    firstIntersectedIntervalCluster);
+        }
+    }
+
+    private void createNewIntervalCluster(Interval<Interval_, Point_> interval) {
+        // Interval does not intersect anything
+        // Ex:
+        //     -----
+        //----       -----
+        var startSplitPoint = splitPointSet.floor(interval.getStartSplitPoint());
+        var newCluster = new IntervalClusterImpl<>(splitPointSet, differenceFunction, startSplitPoint);
+        clusterStartSplitPointToCluster.put(startSplitPoint, newCluster);
+
+        // If there a cluster after this interval, add a new break
+        // between this interval and the next cluster
+        var nextClusterEntry = clusterStartSplitPointToCluster.higherEntry(startSplitPoint);
+        if (nextClusterEntry != null) {
+            var nextCluster = nextClusterEntry.getValue();
+            var difference = differenceFunction.apply(newCluster.getEnd(), nextCluster.getStart());
+            var newBreak = new IntervalBreakImpl<>(newCluster, nextCluster, difference);
+            clusterStartSplitPointToNextBreak.put(startSplitPoint, newBreak);
+        }
+
+        // If there a cluster before this interval, add a new break
+        // between this interval and the previous cluster
+        // (this will replace the old break, if there was one)
+        var previousClusterEntry = clusterStartSplitPointToCluster.lowerEntry(startSplitPoint);
+        if (previousClusterEntry != null) {
+            var previousCluster = previousClusterEntry.getValue();
+            var difference = differenceFunction.apply(previousCluster.getEnd(), newCluster.getStart());
+            var newBreak = new IntervalBreakImpl<>(previousCluster, newCluster, difference);
+            clusterStartSplitPointToNextBreak.put(previousClusterEntry.getKey(), newBreak);
+        }
+    }
+
+    private void removeSpannedBreaksAndUpdateIntersectedBreaks(Interval<Interval_, Point_> interval,
+            IntervalClusterImpl<Interval_, Point_, Difference_> intervalCluster) {
+        var intersectedIntervalBreakMap =
+                clusterStartSplitPointToNextBreak.subMap(
+                        ObjectUtils.defaultIfNull(clusterStartSplitPointToNextBreak.floorKey(interval.getStartSplitPoint()),
+                                interval.getStartSplitPoint()),
+                        true, interval.getEndSplitPoint(), true);
```

Comment:
Another one of those statements that would read better when broken down into component parts.